boards: Add the Nordic nRF5340 audio DK (#49720)
zephyrproject-rtos/zephyr: boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst

.. _nrf5340_audio_dk_nrf5340:

nRF5340 Audio DK
################

Overview
********

The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
with Bluetooth® LE Audio.
Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
for the nRF5340 Audio DK.

> You can use this board for developing LE-Audio-compatible applications that support Auracast™, both the connected isochronous stream (CIS) mode and the broadcast isochronous stream (BIS) modes,  broadcast and unicast, and offer support for both headsets and microphones.

Reviewer: Cut this line into multiple

Hardware
********
The nRF5340 Audio DK comes with the following hardware features:

* nRF5340 dual-core SoC based on the Arm® Cortex®-M33 architecture
* CS47L63 Low-Power Audio DSP with mono differential headphone driver
* nPM1100 Ultra-small form-factor Power Management IC
* On-board digital microphone
* On-board power measurement
* SD card slot
* Built-in debugger
* Stereo analog input using 3.5 mm jack
* USB soundcard capability

.. figure:: img/nrf5340_audio_dk.jpg
     :align: center
     :alt: nRF5340 DK

More information about the board can be found at the `nRF5340 Audio DK website`_.

> nRF5340 SoC

Reviewer: Add newline before new section

===========

> The nRF5340 Audio DK is built around the nRF5340 SoC, which has the following characteristics:

Reviewer: And newline after header

.. _nrf5340_audio_dk_nrf5340:

nRF5340 Audio DK
################

Overview
********

The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
with Bluetooth® LE Audio.

You can use this board for developing LE-Audio-compatible applications that support Auracast™,
connected isochronous streams (CIS) and broadcast isochronous streams (BIS),
and offer support for acting as a audio source, audio sink and source + sink.

Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
for the nRF5340 Audio DK.

Hardware
********

The nRF5340 Audio DK comes with the following hardware features:

* nRF5340 dual-core SoC based on the Arm® Cortex®-M33 architecture
* CS47L63 Low-Power Audio DSP with mono differential headphone driver
* nPM1100 Ultra-small form-factor Power Management IC
* On-board digital microphone
* On-board power measurement
* SD card slot
* Built-in debugger
* Stereo analog input using 3.5 mm jack
* USB soundcard capability

.. figure:: img/nrf5340_audio_dk.jpg
     :align: center
     :alt: nRF5340 DK

> More information about the board can be found at the `nRF5340 Audio DK website`_.

Reviewer 2: The hardware for the Audio DK has been published on Infocenter, so we need to rephrase this part:
```suggestion
More information about the board can be found at the `nRF5340 Audio DK website`_. The `Nordic Semiconductor Infocenter`_
contains the processor's information and the datasheet.
```

Reviewer 3: Fixed

> .. _thingy53_nrf5340:

Reviewer 4: Not a thingy53 :)

Author: Quite picky of you, but OK, I will fix that.

nRF5340 Audio DK
################

Overview
********

The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting

> with Bluetooth LE Audio. Keywords: Auracast™, broadcast, BIS, unicast, CIS, headsets, microphones.

Reviewer 4: Seems redundant both write broadcast and BIS, as well as unicast and CIS. They are, in this context, the same, right?

Reviewer 2: Addressed in my suggestion.

Author: That is true. Added this to match on keyword search.

> The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
> with Bluetooth LE Audio. Keywords: Auracast™, broadcast, BIS, unicast, CIS, headsets, microphones.
> Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
> for the nRF5340 Audio DK. 

Reviewer 2: ```suggestion
The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
with Bluetooth® LE Audio.
Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
for the nRF5340 Audio DK.

You can use this board for developing LE-Audio-compatible applications that support Auracast™, both the connected isochronous stream (CIS) mode and the broadcast isochronous stream (BIS) modes,  broadcast and unicast, and offer support for both headsets and microphones.
```

Hardware
********
* nRF5340 dual-core SoC based on the Arm® Cortex®-M33 architecture
* CS47L63 Low-Power Audio DSP with mono differential headphone driver
* nPM1100 Ultra-small form-factor Power Management IC
* On-board digital microphone
* On-board power measurement
* SD-card
* Built-in debugger
* Stereo analog input using 3.5 mm jack
* USB soundcard capability

.. figure:: img/nrf5340_audio_dk.jpg
     :align: center

>      :alt: nRF5340 DK

Reviewer 2: ```suggestion
     :alt: nRF5340 Audio DK
```

nRF5340 SoC:
============
* A full-featured Arm Cortex-M33F core with DSP instructions,
  FPU, and Armv8-M Security Extension, running at up to 128 MHz, 

>   referred to as the **application core**

Reviewer 4: ```suggestion
  referred to as the **application core**.
```

| GPIO      | on-chip    | gpio                 |
+-----------+------------+----------------------+
| I2C(M)    | on-chip    | i2c                  |
+-----------+------------+----------------------+
| MPU       | on-chip    | arch/arm             |
+-----------+------------+----------------------+
| NVIC      | on-chip    | arch/arm             |
+-----------+------------+----------------------+
| RADIO     | on-chip    | Bluetooth,           |
|           |            | ieee802154           |
+-----------+------------+----------------------+
| RTC       | on-chip    | system clock         |
+-----------+------------+----------------------+
| RTT       | Segger     | console              |
+-----------+------------+----------------------+
| SPI(M/S)  | on-chip    | spi                  |
+-----------+------------+----------------------+
| UARTE     | on-chip    | serial               |
+-----------+------------+----------------------+
| WDT       | on-chip    | watchdog             |
+-----------+------------+----------------------+

Other hardware features have not been enabled yet for this board.
See `Nordic Semiconductor Infocenter`_

> for a complete list of nRF5340 Audio DK board hardware features.

Author: That is currently correct, but the nRF5340 is there. I will alter this section and push a fix today.

Programming and Debugging
*************************

Flashing
========

Follow the instructions in the :ref:`nordic_segger` page to install
and configure all the necessary software. Further information can be
found in :ref:`nordic_segger_flashing`. Then you can build and flash
applications as usual (:ref:`build_an_application` and
:ref:`application_run` for more details).

.. warning::

   The nRF5340 has a flash read-back protection feature. When flash read-back
   protection is active, you will need to recover the chip before reflashing.
   If you are flashing with :ref:`west <west-build-flash-debug>`, run
   this command for more details on the related ``--recover`` option:

   .. code-block:: console

>       west flash -H -r nrfjprog --skip-rebuild

Reviewer 4: Am I missing something, or how is the command here related to the `--recover` option?

Author: It provides information on the --recover option.

Reviewer 4: I see :)
